core/state/journal.go

```diff
@@ -87,6 +90,15 @@ func (ch suicideChange) undo(s *StateDB) {
 	}
 }
 
+var ripemd = common.HexToAddress("0000000000000000000000000000000000000003")
+
+func (ch touchChange) undo(s *StateDB) {
+	if !ch.prev && *ch.account != ripemd {
```

Comment:
Parity has a bad implementation of EIP 161. That caused account `0000000000000000000000000000000000000003` to be deleted in block `2675119`, even though the deletion should have been reverted due to an out of gas error. Geth didn't handle revertals at all (hence today's bug), and because of this, there wasn't a consensus failure 2 days ago. To avoid rewinding the chain, we added this special case for the Parity bug.